### PR TITLE
Lido app/marcus/unique

### PR DESCRIPTION
**Pull request by marhong2**
@tyleradams

### CODE REVIEW

File: formulas/functions/reference.js [hunk 1]
```diff
@@ -343,6 +343,41 @@ const ReferenceFunctions = {
...
+    //Checks to see if range is an array of arrays. If it is not, 
+    //then the first row is Unique to its self and thus can be returned
+    if (typeof range[0] !== "object") {
+      return range;
+    }
+    let seen = new Set();
+    let rv = [];
+    for (let index = 0; index < range.length; index++) {
+      let currValue = range[index];
+      let returnValue = currValue;
```
Comment (tyleradams) [inline formulas/functions/reference.js:367]: If these vars are the same, why do we have both?

File: test/formulas/reference/testcase.js [hunk 2]
```diff
@@ -114,6 +114,30 @@ module.exports = {
...
+        'SUM(UNIQUE(D1:E12))': 20,
+        'SUM(UNIQUE(B2:E10))': 326.77,
+        'SUM(UNIQUE(D1))' : 4,
+        'SUM(UNIQUE(D10:E12))': 11,
+        'SUM(UNIQUE(C2:E2))': 51,
+        'COUNT(UNIQUE(D2:D10))': 1,
+        'COUNT(UNIQUE(B2:C10))': 18,
+        'COUNT(UNIQUE(B2:E10))': 36,
+        'COUNT(UNIQUE(D10:E12))': 4,
+        'COUNT({38,15,25,10,16,20,12})': 7,
```
Comment (tyleradams) [inline test/formulas/reference/testcase.js:130]: Count unique.

File: test/formulas/reference/testcase.js [hunk 3]
```diff
@@ -114,6 +114,30 @@ module.exports = {
         // this should be good, lol
         'SUM(TRANSPOSE({1,2,3;4,5,6}))': 21,
     },
+    
+    UNIQUE: {
+        'SUM(UNIQUE(C2:C10))': 176,
```
Comment (tyleradams) [inline test/formulas/reference/testcase.js:119]: Let's use count, not sum.

**Review by tyleradams — CHANGES_REQUESTED** [review]

File: formulas/functions/reference.js (hunk 1 above)
Comment (marhong2) [reply formulas/functions/reference.js:367]: B/C I modify currValue to run element wise comparisons. So if range[index] is an array, I convert it to a string via .join(). currValue is my comparison var, while returnValue stored the return value

Comment (tyleradams) [reply formulas/functions/reference.js:367]: Ok

File: test/formulas/reference/testcase.js [hunk 4]
```diff
@@ -114,6 +114,31 @@ module.exports = {
...
+        'COUNT(UNIQUE(B2:E10))': 36,
+        'COUNT(UNIQUE(D10:E12))': 4,
+        'COUNT(UNIQUE({38,15,25,10,16,20,12}))': 7,
+        'COUNT(UNIQUE(D1:E12))': 6,
+        'COUNT(UNIQUE(C2:C10))': 8,
+        'COUNT(UNIQUE(B2:C10))': 18,
+        'COUNT(UNIQUE({1;2;3;4;5}))': 5,
+        'SUM(UNIQUE({1;2;3;4;5}))': 15,
+        //duplicates
+        'SUM(UNIQUE({1;1;1;1;1;1;1;1;2;2;2;2;2;3;3;4;4;4;5;5;5;5;6}))': 21,
```
Comment (tyleradams) [inline test/formulas/reference/testcase.js:138]: Count please

**Review by tyleradams — APPROVED** [review]

**Pull request merged**